ClinGen/clincoded
# Connects to #852. Connects to #840. Subtabs and new eval forms

### Connects to #852:
Implemented subtabs into Predictors tab. Also implemented saving of subtab selection into URL.

![image](https://cloud.githubusercontent.com/assets/4326866/17311454/4b317f64-5800-11e6-9bf7-3096d5883f51.png)

*Testing:*

1. Click through subtabs on Predictors tab of VCI
  * Verify that the different subtabs show the correct content
  * Verify that changing subtabs updates the URL properly (first tab will clear the variable in URL)
  * Verify that changing away from Predictors tab properly removes the subtab variable in URL
  * Verify that accessing a subtab via URL works properly

### Connects to #840:
Major overhaul of evaluation forms.

* Evaluation form groups now use a three-column layout
* Code for rendering the form groups are now generated using modularized functions in `form.js`
* Code for Updating and Changing the form values are now generated using modularized ufnctions in `form.js`
* Forms now use dropdowns for status selection
* MAF Cutoff input box was made into un-editable field on curators' request
  * Form select options are filtered depending on what the criteria is
* `form.js` groups now check for conflicting (previously - or -) criteria code statuses upon Save, and displays the appropriate error message
* `form.js` updates evidence data object, instead of creating a new one every time
* `form.js` updated to use new `criteriaStatus` and `criteriaModifier` fields in schema
* Updated `evidence_codes.json`; updated `criteria.js` so it doesn't break w/ new structure
* Implemented `clrAllFormErrors` into global `form.js`
* Implemented error class support into select dropdowns
  * This step required an update in one of the tests
* Various CSS changes/updates/additions

![image](https://cloud.githubusercontent.com/assets/4326866/17311428/29010b58-5800-11e6-8778-019ea2fd3643.png)
![image](https://cloud.githubusercontent.com/assets/4326866/17311446/404240ac-5800-11e6-9285-102a8a81e903.png)

*Limitations/TODOs:*

* `form.js` ALWAYS updates evidence data object, instead of only updating when data was updated. See #740 
* Only the Populations tab was updated to use the new eval forms. See #868

*Testing:*

1. Get to Predictors tab w/ Interpretation
2. Attempt to save evaluation w/ multiple being 'Met' - confirm it does not let you do this w/ proper error messages
3. Try various combinations of values and confirm they work as expected
4. Fix issue and confirm save works
5. Reload page to make sure data retrieval works properly
